### PR TITLE
Parse String and List Slices

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**
Related: #298

### CODE REVIEW

File: src/lpython/parser/tokenizer.re [hunk 1]
```diff
@@ -535,6 +536,7 @@ std::string token2text(const int token)
         T(TK_ATEQUAL, "@=")
         T(TK_RARROW, "->")
         T(TK_COLONEQUAL, ":=")
+        T(TK_DBL_COLON, ":=")
```
Comment (certik) [inline src/lpython/parser/tokenizer.re:539]:
```suggestion
        T(TK_DBL_COLON, "::")
```

**Review by certik — APPROVED** [review]
This looks great. Left some minor comments. I am going to go ahead and merge this, and you can fix those in subsequent PRs.

File: src/lpython/parser/tokenizer.re (hunk 1 above)
Comment (Thirumalai-Shaktivel) [reply src/lpython/parser/tokenizer.re]: Good catch, Thank you for updating the same...

File: src/lpython/parser/parser.yy [hunk 2]
```diff
@@ -578,6 +581,29 @@ async_with_stmt
...
+slice_item
+    : ":"                    { $$ = SLICE_01(nullptr, nullptr, nullptr, @$); }
+    | expr ":"               { $$ = SLICE_01(     $1, nullptr, nullptr, @$); }
+    | ":" expr               { $$ = SLICE_01(nullptr,      $2, nullptr, @$); }
+    | expr ":" expr          { $$ = SLICE_01(     $1,      $3, nullptr, @$); }
+    | "::"                   { $$ = SLICE_01(nullptr, nullptr, nullptr, @$); }
+    | "::" expr              { $$ = SLICE_01(nullptr, nullptr,      $2, @$); }
+    | ":" ":" expr           { $$ = SLICE_01(nullptr, nullptr,      $3, @$); }
+    | expr "::"              { $$ = SLICE_01(     $1, nullptr, nullptr, @$); }
+    | expr ":" ":"           { $$ = SLICE_01(     $1, nullptr, nullptr, @$); }
```
[the remark this answers is not in the document]
Comment (Thirumalai-Shaktivel) [reply src/lpython/parser/parser.yy:597]: Yup, you are correct! I will remove `::` and make the changes in new PR, Thank you sir!

**Pull request merged**